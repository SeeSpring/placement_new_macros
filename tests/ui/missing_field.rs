use placement_new_macros::init;
use std::mem::MaybeUninit;

fn main() {
    struct S {
        a: u32,
    }
    struct T {
        s: S,
        t: [u32; 5],
    }
    fn foo(x: bool) -> u32 {
        x as _
    }
    let mut v = MaybeUninit::<T>::uninit();
    // missing field `t`
    #[rustfmt::skip]
    init!(v, T { s: S { a: f1 } }, f1 = foo(true));
}
