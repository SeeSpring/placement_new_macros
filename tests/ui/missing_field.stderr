error[E0063]: missing field `t` in initializer of `main::T`
  --> $DIR/missing_field.rs:18:5
   |
18 |     init!(v, T { s: S { a: f1 } }, f1 = foo(true));
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ missing `t`
   |
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)
